### PR TITLE
[ci] Switch dependencies serapi <-> coq_lsp

### PR DESCRIPTION
**Pull request by ejgallego**
Now serapi depends on coq_lsp, also serapi tests have been moved there.

To coordinate with the merge of
https://github.com/ejgallego/coq-serapi/pull/409

### CODE REVIEW

**Comment by ejgallego** [comment]
@coqbot run full CI

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -1028,23 +1028,22 @@ library:ci-rupicola:
...
   extends: .ci-template-flambda
 
-plugin:ci-serapi_test:
+plugin:ci-coq_lsp_test:
   extends: .ci-template-flambda
   needs:
   - build:edge+flambda
-  - plugin:ci-elpi_hb
   - library:ci-mathcomp_1
   - plugin:ci-serapi
```
[the remark this answers is not in the document]
Comment (ejgallego) [reply .gitlab-ci.yml:1039]: It does not indeed, sorry I forgot to remove that line (was an oversight, I got distracted with the provenance of the elpi line)

Actually, we removed the math-comp dependency for the tests too, it was painful as we couldn't run the tests on the main Opam CI. I'll remove the coq-lsp test job then and merge it with the main one.

After 8.20 we can just drop serapi entirely from the CI.

File: dev/ci/ci-coq_lsp.sh [hunk 2]
```diff
@@ -11,5 +11,10 @@ if [ "$DOWNLOAD_ONLY" ]; then exit 0; fi
 
 ( cd "${CI_BUILD_DIR}/coq_lsp"
   dune build --root . --only-packages=coq-lsp @install
+  # Tests
   _build/install/default/bin/coq-lsp --version
+  dune runtest test/serlib
```
Comment (SkySkimmer) [inline dev/ci/ci-coq_lsp.sh:16]: should this have `--root .` too?

File: dev/ci/ci-serapi.sh [hunk 3]
```diff
@@ -11,5 +11,7 @@ if [ "$DOWNLOAD_ONLY" ]; then exit 0; fi
 
 ( cd "${CI_BUILD_DIR}/serapi"
   make
-  dune install -p coq-serapi --prefix="$CI_INSTALL_DIR"
+  make test
+  # Not needed by any other CI job for now
+  # dune install -p coq-serapi --prefix="$CI_INSTALL_DIR"
```
Comment (SkySkimmer) [inline dev/ci/ci-serapi.sh:16]: even if it's not needed we should still install I think

Comment (ejgallego) [reply dev/ci/ci-serapi.sh:16]: Ok

File: dev/ci/ci-coq_lsp.sh (hunk 2 above)
Comment (ejgallego) [reply dev/ci/ci-coq_lsp.sh:16]: Yes, thanks for the hint.

**Comment by ejgallego** [comment]
@coqbot run full CI

**Comment by ejgallego** [comment]
Upstream migration has been completed, let's see if this works fine.

I will open another PR to remove SerAPI for after the 8.20 release.

**Comment by ejgallego** [comment]
Oh I removed the overlay branch in error, silly me. Will re-run now with correct branch, this should be merged as soon as possible after.

**Comment by ejgallego** [comment]
@coqbot run full CI

**Comment by ejgallego** [comment]
Relaunched the serapi job that missed the branch here https://github.com/coq/coq/pull/19134

If that works we can merge this even if was rebased (and unbreak the CI)

**Review by SkySkimmer — APPROVED** [review]

**Comment by ejgallego** [comment]
Oh no! We raced with #17393

**Comment by ejgallego** [comment]
@coqbot run full CI

**Comment by SkySkimmer** [comment]
@coqbot merge now

**Pull request merged**